### PR TITLE
client: Implement hide property function

### LINKED ISSUE #28: Implement a function to hide the property on the client
## Feature Request

**Is your feature request related to a problem? Please describe.**  

In order for the client to expose some properties as hash values instead of raw values, a function that maps properties to hash must be implemented.

**Describe the solution you'd like**  

Let's provide a function that takes the names of values the user wants to hide and hash them.
```javascript
export function hideProperty(properties: Property[], hideNames: string[]): Property[] {
  ...
}
```

**Describe alternatives you've considered**  

**Additional context**

### PR DESCRIPTION
**Pull request by HyeockJinKim**
# Summary 

Implement a function that can hide the value by receiving the key of the
property that client want to hide and mapping it with a hash.

- It closes #28 

## How to Review? 

- Please refer to my self-review submitted as the review.

### CODE REVIEW

File: packages/client/test/client.spec.ts [hunk 1]
```diff
@@ -100,3 +100,34 @@ describe('test store localStorage', () => {
...
+    { key: 'address', type: PropertyType.Raw, value: 'Seoul' },
+  ];
+  it('test hide property value', () => {
+    const partiallyHiddenProperties = hideProperty(properties, ['name', 'address']);
+    expect(partiallyHiddenProperties[0].type).to.be.equal(PropertyType.Hash);
+    expect(partiallyHiddenProperties[0].key).to.be.equal('name');
+    expect(partiallyHiddenProperties[0].value).to.be.equal(hashProperty(properties[0]));
+    expect(partiallyHiddenProperties[2].type).to.be.equal(PropertyType.Hash);
+    expect(partiallyHiddenProperties[2].key).to.be.equal('address');
+    expect(partiallyHiddenProperties[2].value).to.be.equal(hashProperty(properties[2]));
```
Comment (HyeockJinKim) [inline packages/client/test/client.spec.ts:117]: Should be hash.

File: packages/client/test/client.spec.ts [hunk 2]
```diff
@@ -100,3 +100,34 @@ describe('test store localStorage', () => {
...
+
+    expect(partiallyHiddenProperties[1].type).to.be.equal(PropertyType.Raw);
+    expect(partiallyHiddenProperties[1].key).to.be.equal('age');
+    expect(partiallyHiddenProperties[1].value).to.be.equal(properties[1].value);
```
Comment (HyeockJinKim) [inline packages/client/test/client.spec.ts:121]: The value must not change.

File: packages/client/test/client.spec.ts [hunk 3]
```diff
@@ -100,3 +100,34 @@ describe('test store localStorage', () => {
...
+  });
+  it('test hidden property verification', () => {
+    const ecdsa = new ECDSA('secp256k1');
+    const key = ecdsa.keyFromPrivate(
+      '2ef40452ec154cd38efdc8ffa52e7f513f7d2b2a77e028342bde96c369e4f77a',
+    );
+    const sign = authPrivacy(properties, { key: key.getPrivate('hex'), type: KeyType.ECDSA });
+    const publicKey = { key: key.getPublic('hex'), type: KeyType.ECDSA };
+    const partiallyHiddenProperties = hideProperty(properties, ['name', 'address']);
+    expect(verifyPrivacy(partiallyHiddenProperties, sign, publicKey)).to.be.equal(true);
```
Comment (HyeockJinKim) [inline packages/client/test/client.spec.ts:131]: Even if the value is hidden, verification should be possible.

File: packages/server/test/auth.spec.ts [hunk 4]
```diff
@@ -1,34 +1,16 @@
...
-      value: 'ba7816bf8f01cfea414140de5dae2223b00361a396177a9cb410ff61f20015ad',
-    };
-
-    expect(hashProperty(property1)).to.be.equal('2e832436ce5b19a381949a13547469604758fd5e0001206f8a6cf0ed7974145a');
-    expect(hashProperty(property1)).to.be.not.equal(hashProperty(property2));
-    expect(hashProperty(property3)).to.be.equal(hashProperty(property3));
-  });
-  it('test hash', () => {
-    expect(hash('abc')).to.be.equal('ba7816bf8f01cfea414140de5dae2223b00361a396177a9cb410ff61f20015ad');
-  });
```
Comment (anthonyminyungi) [inline packages/server/test/auth.spec.ts:31]: Moved to `packages/message/test/message.spec.ts`.

File: packages/server/src/utils.ts [hunk 5]
```diff
@@ -1,23 +1,8 @@
...
-export function hash(msg: string): string {
-  return sha256(msg).toString();
-}
-
-export function hashProperty(property: Property): string {
-  if (property.type === PropertyType.Hash) return property.value;
-
-  return hash(`${utf8ToBase64(property.key)}:${utf8ToBase64(property.value)}`);
-}
-
```
Comment (anthonyminyungi) [inline packages/server/src/utils.ts:20]: Moved to `packages/message/src/index.ts`.

File: packages/client/test/client.spec.ts [hunk 6]
```diff
@@ -47,7 +47,7 @@ describe('test utils', () => {
     ).to.be.equal('Message');
   });
   it('test Encrypt & Decrypt Long message', () => {
-    const message = "{'properties':[{'key':'name','type':0,'value':'Kim'},{'key':'age','type':0,'value':'17'},{'key':'address','type':0,'value':'Seoul'}],'sign':{'authType':0,'keyType':1,'value':'304402207282d176a100f0d5feb3faa160bd39843253ec98487cc3342905260ab592b85e02201fdf55464c12bbe4b93868a6e84da59e816e918d52cc599678605f61846426c6'}}\n";
+    const message = '{\'properties\':[{\'key\':\'name\',\'type\':0,\'value\':\'Kim\'},{\'key\':\'age\',\'type\':0,\'value\':\'17\'},{\'key\':\'address\',\'type\':0,\'value\':\'Seoul\'}],\'sign\':{\'authType\':0,\'keyType\':1,\'value\':\'304402207282d176a100f0d5feb3faa160bd39843253ec98487cc3342905260ab592b85e02201fdf55464c12bbe4b93868a6e84da59e816e918d52cc599678605f61846426c6\'}}\n';
```
Comment (anthonyminyungi) [inline packages/client/test/client.spec.ts:50]: How about just reversing them?
```suggestion
    const message = '{"properties":[{"key":"name","type":0,"value":"Kim"}
    ...}\n';
```

File: packages/server/src/index.ts [hunk 7]
```diff
@@ -26,7 +23,7 @@ export function authPrivacy(
 ): Signature {
   const hashes = properties.map((property) => hashProperty(property));
   const merkleRoot = getMerkleRoot(hashes);
-  return signAccordingToKeyType(merkleRoot, secret.key, secret.type);
+  return signByKeyType(merkleRoot, secret.key, secret.type);
```
Comment (anthonyminyungi) [inline packages/server/src/index.ts:26]: Good reflection!

**Pull request merged**